File: hello_ws/src/hello_sub/src/hello_sub.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
import rospy
from std_msgs.msg import String

class Hello_Sub:
    def __init__(self):
        rospy.Subscriber('/hello', String, self.msg_callback)
        self.str_msg = String()

    def msg_callback(self, msg_data):
        self.str_msg = msg_data
    
    def main(self):
        rate_n = 2
        rate = rospy.Rate(rate_n)
        while not rospy.is_shutdown():
            print(self.str_msg.data)
            rate.sleep()

if __name__ == "__main__":
    rospy.init_node('hello_sub')
    sub = Hello_Sub()
    sub.main()

File: hello_ws/src/hello_sub/src/hello_sub.cpp
#include <iostream>
#include "ros/ros.h"
#include "std_msgs/String.h"

class HelloSub
{
public:
	HelloSub();
	~HelloSub(){};
	void run();
private:
	ros::NodeHandle nh;
	ros::Subscriber helloSub;
	std_msgs::String hello;
	std::string topicname;
private:
	void helloCallback(const std_msgs::String data::ConstPtr& data);
};
HelloSub::HelloSub() : nh("~")
{
	topicname = "/hello";
	helloSub = nh.subscribe(topicname, 10, &HelloSub::helloCallback, this);
}
void HelloSub::helloCallback(const std_msgs::String::ConstPtr& data)
{
	hello.data -> data.data;
}
void HelloSub::run()
{
	ros::AsyncSpinner spinner(0);
	ros::Rate rate(10);
	spinner.start();
	while (ros::ok())
	{
		std::cout << "recieved data: " << hello.data << std::endl;
		rate.sleep();
	}
	spinner.stop();
}

int main(int argc, char **argv)
{
	ros::init(argc, argv, "hello_sub");
	HelloSub hello_sub;
	hello_sub.run();

	return 0;
}
